packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1640,6 +1643,23 @@ class SelectableRegionState extends State<SelectableRegion>
     _finalizeSelectableRegionStatus();
   }
 
+  void _granularlyHorizontalExtendSelection(TextGranularity granularity, bool right) {
+    _directionalHorizontalBaseline = null;
+    if (!_selectionDelegate.value.hasSelection) {
+      return;
+    }
+    _selectable?.dispatchSelectionEvent(
+      HorizontalGranularlyExtendSelectionEvent(
```

Comment:
So we don't know the text direction at this point because this is the whole SelectableRegionState, which internally could have different TextDirection widgets, such as with my example app (in https://github.com/flutter/flutter/pull/160516#pullrequestreview-2560045733).